contracts/token/ERC721/ERC721.sol

```diff
@@ -331,8 +331,24 @@ contract ERC721 is Context, ERC165, IERC721 {
             return true;
         }
 
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSignature(
+            "onERC721Received(address,address,uint256,bytes)",
+            msg.sender,
+            from,
+            tokenId,
+            _data
+            );
+
+            //solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory returndata) = to.call(payload);
+        if (!success) {
+            if (returndata.length > 0){
+                revert(string(returndata));
+            }else 
+                revert("ERC721: to address does not implement ERC721Received interface");
+        } else {return true;} 
+        // bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
+        // return (retval == _ERC721_RECEIVED);
```

Comment:
Please tidy up the whitespace here to conform to our usual style and remove the commented out lines.  :slightly_smiling_face: